### PR TITLE
Bump to 8.1.4-snapshot / clear some warnings.

### PR DESCRIPTION
**Pull request by natechadwick**

### CODE REVIEW

File: designer/core/src/main/java/com/percussion/client/models/impl/PSHierarchyManager.java [hunk 1]
```diff
@@ -208,7 +215,7 @@ public IPSHierarchyNodeRef[] cloneChildren(
       }
       
       IPSHierarchyNodeRef[] validRefsArray = 
-         validRefs.toArray(new IPSHierarchyNodeRef[validRefs.size()]);
+         validRefs.toArray(new IPSHierarchyNodeRef[0]);
```
Comment (sonydeswal74) [inline designer/core/src/main/java/com/percussion/client/models/impl/PSHierarchyManager.java:218]: It was setting it to validRefs.size() and now set to 0...

File: designer/core/src/main/java/com/percussion/client/models/impl/PSHierarchyManager.java [hunk 2]
```diff
@@ -310,7 +317,7 @@ public IPSHierarchyNodeRef[] createChildren(
...
```
Comment (sonydeswal74) [inline designer/core/src/main/java/com/percussion/client/models/impl/PSHierarchyManager.java:320]: Same as above

File: designer/core/src/main/java/com/percussion/client/models/impl/PSHierarchyManager.java [hunk 3]
```diff
@@ -386,7 +393,7 @@ private void saveObjects(Object[] refs, Object[] data, boolean foldersOnly)
             return;
          //go directly to proxy, don't use our model layer
          m_model.getProxy().save(
-               folderRefs.toArray(new IPSReference[folderRefs.size()]),
+               folderRefs.toArray(new IPSReference[0]),
```
Comment (sonydeswal74) [inline designer/core/src/main/java/com/percussion/client/models/impl/PSHierarchyManager.java:396]: Same as above

File: designer/core/src/main/java/com/percussion/client/models/impl/PSHierarchyManager.java [hunk 4]
```diff
@@ -430,40 +437,38 @@ private NodeId createId(IPSHierarchyNodeRef ref)
...
+         List<IPSHierarchyNodeRef> delete = new ArrayList<>();
+         for (IPSHierarchyNodeRef hRef : children) {
             if (hRef.isPersisted())
                delete.add(hRef);
          }
          if (!delete.isEmpty())
          {
             IPSHierarchyNodeRef[] childRefs = delete
-               .toArray(new IPSHierarchyNodeRef[delete.size()]);
+               .toArray(new IPSHierarchyNodeRef[0]);
```
Comment (sonydeswal74) [inline designer/core/src/main/java/com/percussion/client/models/impl/PSHierarchyManager.java:453]: Same as above

File: designer/core/src/main/java/com/percussion/client/models/impl/PSHierarchyManager.java [hunk 5]
```diff
@@ -430,40 +437,38 @@ private NodeId createId(IPSHierarchyNodeRef ref)
...
             getProxy().removeChildren(childRefs);
          }
          m_model.objectsDeleted(children
-            .toArray(new IPSHierarchyNodeRef[children.size()]));
+            .toArray(new IPSHierarchyNodeRef[0]));
```
Comment (sonydeswal74) [inline designer/core/src/main/java/com/percussion/client/models/impl/PSHierarchyManager.java:457]: SAme

File: vedit/vedit.ui/pom.xml [hunk 6]
```diff
@@ -21,7 +21,7 @@
     <packaging>eclipse-plugin</packaging>
     <parent>
         <groupId>com.percussion</groupId>
-        <version>8.1.3</version>
+        <version>8.1.4-SNAPSHOT</version>
```
Comment (sonydeswal74) [inline vedit/vedit.ui/pom.xml:24]: why are we adding SNAPSHOT when our final jar is only 8.1.4?

**Review by sonydeswal74 — COMMENTED** [review]
Just went over the task.. just have few questions. rest looks good.

File: designer/core/src/main/java/com/percussion/client/models/impl/PSHierarchyManager.java (hunk 1 above)
Comment (natechadwick) [reply designer/core/src/main/java/com/percussion/client/models/impl/PSHierarchyManager.java:218]: It's a performance thing: https://stackoverflow.com/questions/53284214/toarray-with-pre-sized-array

File: vedit/vedit.ui/pom.xml (hunk 6 above)
Comment (natechadwick) [reply vedit/vedit.ui/pom.xml:24]: Because it is still in development, when we release we remove the snapshot.

**Review by sonydeswal74 — APPROVED** [review]

**Pull request merged**